app/v1/services/RetrieveBusinessDetailsService.scala

```diff
@@ -64,15 +64,24 @@ class RetrieveBusinessDetailsService @Inject() (connector: RetrieveBusinessDetai
     }
   }
 
-  private val downstreamErrorMap: Map[String, MtdError] =
-    Map(
+  private val downstreamErrorMap: Map[String, MtdError] = {
+    val errors = Map(
       "INVALID_NINO"        -> NinoFormatError,
       "INVALID_MTDBSA"      -> InternalError,
       "NOT_FOUND_NINO"      -> NotFoundError,
       "NOT_FOUND_MTDBSA"    -> InternalError,
-      "INVALID_IDTYPE"      -> InternalError,
       "SERVER_ERROR"        -> InternalError,
       "SERVICE_UNAVAILABLE" -> InternalError
     )
 
+    val extraIfsErrors = Map(
+      "INVALID_MTD_ID"        -> InternalError,
+      "INVALID_CORRELATIONID" -> InternalError,
+      "INVALID_IDTYPE"        -> InternalError,
+      "UNMATCHED_STUB_ERROR"  -> RuleIncorrectGovTestScenarioError,
```

Comment:
This can be put under `val errors` instead as it isn't specific to IFS errors. For all files

Reply:
fixed, please see latest code